Source/Core/UICommon/NetPlayIndex.cpp

```diff
@@ -0,0 +1,297 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "UICommon/NetPlayIndex.h"
+
+#include <picojson/picojson.h>
+
+#include "Common/HttpRequest.h"
+#include "Common/Thread.h"
+#include "Common/Version.h"
+
+#include "Core/Config/NetplaySettings.h"
+
+NetPlayIndex::NetPlayIndex()
+{
+}
+
+NetPlayIndex::~NetPlayIndex()
+{
+  if (!m_secret.empty())
+    Remove();
+}
+
+static std::optional<picojson::value> ParseResponse(std::optional<std::vector<u8>> response)
+{
+  std::string response_string(reinterpret_cast<char*>(response->data()), response->size());
+
+  picojson::value json;
+
+  auto error = picojson::parse(json, response_string);
+
+  if (!error.empty())
+    return {};
+
+  return json;
+}
+
+std::optional<std::vector<NetPlaySession>>
+NetPlayIndex::List(const std::map<std::string, std::string>& filters)
+{
+  std::vector<NetPlaySession> sessions;
+
+  Common::HttpRequest request;
```

Comment:
These variables can be declared right before they are used.

Reply:
It is used in the block immediately following the next variable definition (``request.EscapeComponent``).

Reply:
Ah, `request` is alright. `sessions` isn't used until that final loop though it looks like.